app/src/display/main.c

```diff
@@ -32,19 +33,43 @@ void display_tick_cb(struct k_work *work) { lv_task_handler(); }
 
 K_WORK_DEFINE(display_tick_work, display_tick_cb);
 
+#if IS_ENABLED(CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED)
+
+K_THREAD_STACK_DEFINE(display_work_stack_area, CONFIG_ZMK_DISPLAY_DEDICATED_THREAD_STACK_SIZE);
+
+static struct k_work_q display_work_q;
+
+struct k_work_q *zmk_display_work_q() {
```

Comment:
Since the function signature doesn't depend on CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED, this might be a little easier to maintain if you made it be 
```c
struct k_work_q *zmk_display_work_q() {
#if IS_ENABLED(CONFIG_ZMK_DISPLAY_WORK_QUEUE_DEDICATED)
    return &display_work_q;
#else
    return &k_sys_work_q;
#endif
}
```

Reply:
Wouldn't this approack create a `static struct k_work_q display_work_q;` in both `if` branches?

Reply:
I'm not sure I understand what you mean, but probably no? You would statically define the work queue in a separate #if block, or you could even define it inside the function itself if there wasn't any code outside the function that needed to reference it.

Reply:
I'm implementing a simular solution for the mousekeys PR, so the question just kind of sprung up, and I thought of the separate `#if` block right after the question. Thanks! This looks a tad more elegant.

Reply:
Yeah, I like that refactor to avoid the duplication of the function signature. Pushed that.